core/src/main/java/hudson/init/impl/InstallUncaughtExceptionHandler.java

```diff
@@ -17,31 +18,27 @@
 import org.kohsuke.stapler.Stapler;
 
 /**
- * @author Kohsuke Kawaguchi
+ * Deals with exceptions that get thrown all the way up to the Stapler rendering layer.
  */
 public class InstallUncaughtExceptionHandler {
 
     private static final Logger LOGGER = Logger.getLogger(InstallUncaughtExceptionHandler.class.getName());
 
     @Initializer
     public static void init(final Jenkins j) throws IOException {
-        CompressionFilter.setUncaughtExceptionHandler(j.servletContext, new UncaughtExceptionHandler() {
-            @Override
-            public void reportException(Throwable e, ServletContext context, HttpServletRequest req, HttpServletResponse rsp) throws ServletException, IOException {
+        CompressionFilter.setUncaughtExceptionHandler(j.servletContext, (e, context, req, rsp) -> {
                 if (rsp.isCommitted()) {
-                    LOGGER.log(Level.WARNING, null, e);
+                    LOGGER.log(isEOFException(e) ? Level.FINE : Level.WARNING, null, e);
                     return;
                 }
                 req.setAttribute("javax.servlet.error.exception",e);
                 try {
-                    WebApp.get(j.servletContext).getSomeStapler()
-                            .invoke(req,rsp, Jenkins.getInstance(), "/oops");
+                    WebApp.get(j.servletContext).getSomeStapler().invoke(req, rsp, Jenkins.get(), "/oops");
```

Comment:
Unrelated but I love this details.

Reply:
I was reindenting this block anyway, so cleaned up formatting and a deprecated warning while it was “free”.